### PR TITLE
refactor: groups

### PR DESCRIPTION
**Pull request by spider0061**
## Description
- add relevant comments
- remove unnecessary white spaces
- add white spaces whereever requrired
- add groups resolver

## Related issues and discussion
#643 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/groups/groups.component.ts [hunk 1]
```diff
@@ -16,6 +21,7 @@ export class GroupsComponent implements OnInit {
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
+
   constructor(private groupService: GroupsService) { }
 
   ngOnInit() {
```
Comment (abhi40308) [inline src/app/groups/groups.component.ts]: remove console.log inside this

File: src/app/groups/groups.component.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import { Component, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
```
Comment (Rishabh106) [inline src/app/groups/groups.component.ts]: 'Angular Imports' comment should be added here.

File: src/app/groups/groups.component.ts [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
 import {MatPaginator, MatSort, MatTableDataSource} from '@angular/material';
-import { GroupsService } from './groups.service';
 import {DataSource} from '@angular/cdk/collections';
```
Comment (Rishabh106) [inline src/app/groups/groups.component.ts]: Indentation between { } should also be fixed.

**Comment by karantakalkar** [comment]
@spider0061 can you make the groups data pass through a resolver, also do the same for centers refactor PR.

**Comment by spider0061** [comment]
@karantakalkar I'll do it

**Comment by spider0061** [comment]
> @spider0061 can you make the groups data pass through a resolver, also do the same for centers refactor PR.

@karantakalkar Have a look. If it is fine then I will do it for centers too or let me know if there are changes required.

**Comment by karantakalkar** [comment]
@spider0061 this PR has an overlap with #507.
 @abhaychawla can you opine on this, please.

**Comment by karantakalkar** [comment]
also sorry about wrong suggestion of using resolver, I didn't foresee that a custom data source will be used

**Comment by spider0061** [comment]
@karantakalkar I'll change it. @abhaychawla Sorry I didn't look at Punwai's PR at that time. Do let me know what should I do now.

**Comment by abhaychawla** [comment]
@spider0061, as there was already an existing PR, this will be closed as duplicate.

**Pull request closed without merging**